tests/core/util/test_datetime_helpers.py

```diff
@@ -154,3 +154,137 @@ def test_boundaries(self, start, until, months):
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 3, 1),
+                5,
+            ),
+            (
+                datetime_utc(2000, 4, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 4, 1),
+                6,
+            ),
+            (
+                datetime_utc(2000, 5, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 5, 1),
+                7,
+            ),
+            (
+                datetime_utc(2000, 6, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 6, 1),
+                8,
+            ),
+            (
+                datetime_utc(2000, 7, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 7, 1),
+                9,
+            ),
+            (
+                datetime_utc(2000, 8, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 8, 1),
+                10,
+            ),
+            (
+                datetime_utc(2000, 9, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 9, 1),
+                11,
+            ),
+            (
+                datetime_utc(2000, 10, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 10, 1),
+                12,
+            ),
+            (
+                datetime_utc(2000, 11, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 11, 1),
+                13,
+            ),
+            (
+                datetime_utc(2000, 12, 15),
+                datetime.date(1999, 10, 1),
+                datetime.date(2000, 12, 1),
+                14,
+            ),
+            (
+                datetime_utc(2000, 12, 15),
+                datetime.date(2000, 6, 1),
+                datetime.date(2000, 12, 1),
+                6,
+            ),
+            (
+                datetime_utc(2000, 12, 15),
+                datetime.date(2000, 6, 1),
+                datetime.date(2000, 12, 1),
+                6,
+            ),
+            (
+                datetime_utc(2000, 12, 15),
+                datetime.date(1999, 6, 1),
+                datetime.date(2000, 12, 1),
+                18,
+            ),
+            (
+                datetime_utc(2000, 12, 15),
+                datetime.date(1990, 6, 1),
+                datetime.date(2000, 12, 1),
+                126,
+            ),
+            (
+                datetime_utc(2000, 12, 15),
+                datetime.date(1999, 12, 1),
+                datetime.date(2000, 12, 1),
+                12,
+            ),
+        ],
+    )
+    def test_boundaries_at_different_current_times(
+        self, current_datetime, start, until, months
+    ):
+        with patch("core.util.datetime_helpers.utc_now") as mock_utc_now:
+            mock_utc_now.return_value = current_datetime
```

Comment:
The test cases covered in `test_boundaries` (above) are fully replicated here in `test_boundaries_at_different_current_times`, so I will remove `test_boundaries`.